app/models/solidus_afterpay/payment_method.rb

```diff
@@ -33,24 +31,39 @@ def try_void(payment)
     end
 
     def available_for_order?(order)
-      available_payment_currency == order.currency && available_payment_range.include?(order.total)
+      currency == order.currency && available_payment_range.include?(order.total)
     end
 
     private
 
     def available_payment_range
-      min = preferred_minimum_amount || configuration&.minimumAmount&.amount.to_f
-      max = preferred_maximum_amount || configuration&.maximumAmount&.amount.to_f
+      minimum_amount..maximum_amount
+    end
 
-      min..max
+    def configuration
+      @configuration ||= gateway.retrieve_configuration
     end
 
-    def available_payment_currency
-      preferred_currency || configuration&.maximumAmount&.currency
+    def minimum_amount
+      Rails.cache.fetch("solidus_afterpay_min_amount", expires_in: expires_in_seconds) do
+        configuration&.minimumAmount&.amount.to_f
+      end
     end
 
-    def configuration
-      @configuration ||= gateway.retrieve_configuration
+    def maximum_amount
+      Rails.cache.fetch("solidus_afterpay_max_amount", expires_in: expires_in_seconds) do
+        configuration&.maximumAmount&.amount.to_f
+      end
+    end
+
+    def currency
```

Comment:
why did you rename this method from `available_payment_currency` to `currency`?

Reply:
I changed it back!